tests/minigraph/test_masked_services.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import time
+from tests.common.helpers.assertions import pytest_assert
+
+def is_service_loaded(duthost, service):
+    output = duthost.shell("systemctl show -p LoadState --value {}.service".format(service))
+    if output['rc'] != "0":
+        pytest.fail("Error showing load state of telemetry service")
+        return False
+    return output['stdout'] == "loaded"
+
+def change_service_state(duthost, service, enable):
+    outputs = []
+    if enable:
+        outputs = [
+            duthost.shell("systemctl unmask {}.service".format(service)),
+            duthost.shell("systemctl enable {}.service".format(service)),
+            duthost.shell("systemctl start {}.service".format(service))
+        ]
+    else:
+        outputs = [
+            duthost.shell("systemctl stop {}.service".format(service)),
+            duthost.shell("systemctl disable {}.service".format(service)),
+            duthost.shell("systemctl mask {}.service".format(service))
+        ]
+    for output in outputs:
+        if output['rc'] != 0:
+            pytest.fail("Error starting or stopping service")
+            return
+
+def test_masked_services(duthosts, rand_one_dut_hostname):
+    """
+    @summary: This test case will mask telemetry service, then test load_minigraph and check its success
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    test_service = "telemetry"
+
+    logging.info("Bringing down telemetry service")
+    service_status = duthost.critical_process_status("telemetry")
+    if not service_status:
+        pytest.fail("Make sure telemetry is up and running before calling this test")
+
+    change_service_state(duthost, test_service, False)
+    logging.info("Wait until service is masked and inactive")
+
+    time.sleep(3) #give adequate time for service to become masked and inactive
```

Comment:
>sleep [](http://example.com/codeflow?start=9&length=5)

`sleep` has some disadvantages
1. if too long, it will waste time because the waiting situation mostly likely come earlier
2. if too short, it will make testcase flaky.

Suggest use [wait_until](https://github.com/sonic-net/sonic-mgmt/blob/2e40f5068803da7d0f8382a690faa124b3bb59be/tests/common/utilities.py#L85)

Reply:
Done